config/v1/types_oauth.go

```diff
@@ -139,50 +141,16 @@ const (
 	// that user name already exists, the identity is mapped to the existing user, adding to any
 	// existing identity mappings for the user.
 	MappingMethodAdd MappingMethodType = "add"
-
-	// MappingMethodGenerate provisions a user with the identity’s preferred user name. If a user
-	// with the preferred user name is already mapped to an existing identity, a unique user name is
-	// generated, e.g. myuser2. This method should not be used in combination with external
-	// processes that require exact matches between openshift user names and the idp user name
-	// such as LDAP group sync.
-	MappingMethodGenerate MappingMethodType = "generate"
```

Comment:
if @ericavonb signs off on the extra migration pain (two releases I think) and you keep this as an option to your binary so that if a customer uses it they aren't completely stuck and you have a metric so we know how many people are using this option directly on the binary, then removing the option here can work.

Reply:
across major versions, I do not have the same concern that @deads2k is expressing.  we can add this back if required.